storage/src/test/java/io/atomix/storage/journal/AbstractJournalTest.java

```diff
@@ -91,6 +92,47 @@ public static Collection primeNumbers() {
         .build();
   }
 
+  @Test
+  public void shouldBeEmpty() {
+    // given
+    final Journal<TestEntry> journal = createJournal();
+    final JournalReader<TestEntry> reader = journal.openReader(1, Mode.ALL);
+
+    // when
+    final boolean empty = reader.isEmpty();
+
+    // then
+    assertTrue(empty);
+  }
+
+  @Test
+  public void shouldNotBeEmpty() {
+    // given
+    final Journal<TestEntry> journal = createJournal();
+    final JournalReader<TestEntry> reader = journal.openReader(1, Mode.ALL);
+
+    // when
+    journal.writer().append(new TestEntry(8));
+    final boolean empty = reader.isEmpty();
+
+    // then
+    assertFalse(empty);
+  }
+
+  @Test
+  public void shouldBeEmptyIfNothingCommitted() {
```

Comment:
Please add an test that it is no longer empty when you commit the entry

Reply:
Oops, I was sure I had added that one :thinking: